### ISSUE TITLE
False positive error report E1101 with EnvironmentError

### ISSUE THREAD

**Issue by plocharz-9livesdata** [PyCQA/pylint#2777]
`test-pylint.py`
```python
import os
import psutil

try:
    os.stat("does not exist")
except psutil.Error as e:
    pass

try:
    os.stat("does not exist")
except EnvironmentError as e:
    err = e.strerror
```
### Steps to reproduce
1. run `pylint -E test-pylint.py`

### Current behavior
```
************* Module test-pylint
test-pylint.py:12:10: E1101: Instance of 'Error' has no 'strerror' member (no-member)
```

### Expected behavior
No output is expected

### Workaround:
Use different names for error variables:
```python
import os
import psutil

try:
    os.stat("does not exist")
except psutil.Error as e:
    pass

try:
    os.stat("does not exist")
except EnvironmentError as exc:
    err = exc.strerror
```

### pylint --version output
```
pylint 2.3.0
astroid 2.2.0
Python 3.6.7 (v3.6.7:6ec5cf2, Oct 22 2018, 11:29:07) 
[GCC 4.8.4]
```

**Comment by PCManticore** [comment]
Yikes, thanks for reporting the issue! What's happening is that the inference engine is dumb and can't limit the inference of `e` from the second `except` handler, resulting in inferring both `e` as both `Error` and `EnvironmentError`. We should definitely address this in `astroid`.

**Comment by PCManticore** [comment]
This is fixed in astroid's master which is going to be astroid 2.3.0 when launched.
